src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -67,10 +67,10 @@
      (handler (assoc request :jruby-instance jruby-instance)))))
 
 (defn get-environment-from-request
-  "Gets the environment from a request."
+  "Gets the environment from the URL or query string of a request."
   [req]
-  (-> req
-      (get-in [:params "environment"])))
+  (or (get-in req [:params :environment])
+      (get-in req [:params "environment"])))
```

Comment:
I don't have any concerns, except for the symbol vs string shenanigans of puppet past.